noir-projects/noir-contracts/contracts/docs_example_contract/src/main.nr

```diff
@@ -266,7 +266,8 @@ contract DocsExample {
     // docs:end:private_mutable_is_initialized
 
     // docs:start:get_note-private-immutable
-    unconstrained fn get_imm_card() -> pub CardNote {
+    #[aztec(private)]
+    fn get_imm_card() -> CardNote {
         storage.private_immutable.get_note()
     }
```

Comment:
This was wrong: we can't call `get_note` from `unconstrained` since it requires a private context to store side-effects in! This function always fails when called, but no test called it so we never noticed this.

The fix is straightforward, though it did force me to implement the `Serialize` trait for `CardNote` now that we return it from a private function.